python/ml4ir/base/data/tfrecord_reader.py

```diff
@@ -797,7 +797,7 @@ def read(
         # Parallel calls set to AUTOTUNE: improved training performance by 40% with a classification model
         dataset = (
             dataset.map(parse_fn, num_parallel_calls=tf.data.experimental.AUTOTUNE)
-            # .apply(data.experimental.ignore_errors())
+             .apply(data.experimental.ignore_errors())
```

Comment:
Can you share some context on why this was commented and why we are now uncommenting? 

LGTM though. 

A separate issue: 
- On the creation of the dataset here, are we applying any randomization? In other words does the model see the data always in the same order or not? I am looking at this [for example](https://www.tensorflow.org/api_docs/python/tf/data/Dataset#shuffle)

Reply:
I'm uncommenting this line to skip bad queries without throwing an error.

Offline, I believe the model see the data in the same order. However, running on Flowsnake, I believe there is some "un-planned" shuffling that happens perhaps due to spark.

Reply:
FWIW, I think shuffling the data per epoch would produce a better model.

Reply:
Do you want to add it?

Reply:
+1 on shuffling